src/adapters/AbstractRichtextEditorAdapter.ts

```diff
@@ -224,12 +224,17 @@ export abstract class AbstractRichtextEditorAdapter implements AdapterInterface
     assertElementIsDisplayed(this.getEditorElement());
     const [alignedMatches] = this.selectMatches(checkId, matchesWithReplacement);
     const replacement = alignedMatches.map((m) => m.originalMatch.replacement).join('');
+
+    // Capturing parent node is necessary, replacement makes the textnode orphan.
+    const parentNode = this.getEditorDocument().getSelection()?.focusNode?.parentNode;
```

Comment:
What happens if the parentNode is undefined?

Reply:
@loaiabdelhalim It will not trigger the input event and log a warning message to the browser console. https://github.com/acrolinx/sidebar-sdk-js/pull/91/files#diff-5acab41990585fa68ae01ff7b2e2915628d19237840c3061b31fcadc1c0f09b8R49

Reply:
Fixed. Input simulation will be triggered only if parent node is defined